### PR TITLE
Update to Gradle 7.6

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Update to Gradle 7.6

### Issues Resolved
N/A

### Testing
N/A (build tooling)

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by DarshitChanpura — COMMENTED** [review]
@reta What do you think about running `./gradlew wrapper --gradle-version 7.6 --distribution-type all` ? This will update all files under gradle/wrapper folder

**Comment by reta** [comment]
> @reta What do you think about running `./gradlew wrapper --gradle-version 7.6 --distribution-type all` ? This will update all files under gradle/wrapper folder

The wrapper didn't change :)

File: gradlew [hunk 1]
```diff
@@ -1,18 +1,7 @@
...
-# this file be licensed under the Apache-2.0 license or a
-# compatible open source license.
-#
-# Modifications Copyright OpenSearch Contributors. See
-# GitHub history for details.
-#
-
-#
-# Copyright 2015-2021 the original author or authors.
+# Copyright © 2015-2021 the original authors.
```
Comment (DarshitChanpura) [inline gradlew:4]: We should re-add these licenses. Thoughts?

Comment (reta) [reply gradlew:4]: Definitely not, those are 100% from Gradle, we do not own them or modified them

Comment (reta) [reply gradlew:4]: May be during the migration, or they were modified, ...

**Comment by reta** [comment]
The `build-artifact-names` is failing with :+1: 

```
* What went wrong:* 
A problem occurred evaluating root project 'opensearch-security'.
> 'org.gradle.internal.jvm.inspection.JvmInstallationMetadata org.gradle.internal.jvm.inspection.JvmMetadataDetector.getMetadata(java.io.File)'
```

It seems like we have very old 3.0.0 core snapshots (https://build.ci.opensearch.org/job/distribution-build-opensearch/) :(

**Review by cwperks — DISMISSED** [review]

**Review by peternied — APPROVED** [review]

**Comment by peternied** [comment]
Looks like new gradle didn't want to run the increment version task / test

> FAILURE: Build failed with an exception.
> 
> * Where:
> Build file '/home/runner/work/security/security/build.gradle' line: 68
> 
> * What went wrong:
> A problem occurred evaluating root project 'opensearch-security'.
> > 'org.gradle.internal.jvm.inspection.JvmInstallationMetadata org.gradle.internal.jvm.inspection.JvmMetadataDetector.getMetadata(java.io.File)'

**Comment by reta** [comment]
> Looks like new gradle didn't want to run the increment version task / test
> 
> > FAILURE: Build failed with an exception.
> > 
> > * Where:
> >   Build file '/home/runner/work/security/security/build.gradle' line: 68
> > * What went wrong:
> >   A problem occurred evaluating root project 'opensearch-security'.
> > 
> > > 'org.gradle.internal.jvm.inspection.JvmInstallationMetadata org.gradle.internal.jvm.inspection.JvmMetadataDetector.getMetadata(java.io.File)'

Correct @peternied , we need core builds (https://github.com/opensearch-project/security/pull/2310#issuecomment-1343099392), working on that

**Review by DarshitChanpura — APPROVED** [review]
Thank you @reta!

**Comment by DarshitChanpura** [comment]
I will merge this now as the failure is unrelated and will be tracked separately

**Comment by peternied** [comment]
> I will merge this now as the failure is unrelated and will be tracked separately

Can you link the issue we are using to track this?

**Comment by DarshitChanpura** [comment]
> > I will merge this now as the failure is unrelated and will be tracked separately
> 
> Can you link the issue we are using to track this?

done. https://github.com/opensearch-project/security/pull/2310#issuecomment-1343294180

**Comment by reta** [comment]
Thanks a lot @peternied @DarshitChanpura , I've added the clarification what is the problem, the issue should go away once newer OpenSearch core 3.0.0 bits are published, no actions are required on the `security` plugin side.

**Pull request merged**